src/tests.ts

```diff
@@ -1,11 +1,82 @@
+import * as os from "os";
 import * as vscode from "vscode";
-import { ExecResult, exec, extensionConfiguration } from "./utils";
-import { Tests, DebugEnvironmentConfiguration } from "./types";
+import { ExecResult, exec, extensionConfiguration, getTargetName } from "./utils";
+import { Target, Targets, Test, Tests, DebugEnvironmentConfiguration } from "./types";
 import { getMesonTests, getMesonTargets } from "./introspection";
 import { workspaceState } from "./extension";
 
-export async function rebuildTests(controller: vscode.TestController) {
-  let tests = await getMesonTests(workspaceState.get<string>("mesonbuild.buildDir")!);
+// this is far from complete, but should suffice for the
+// "test is made of a single executable is made of a single source file" usecase.
+function findSourceOfTest(test: Test, targets: Targets): vscode.Uri | undefined {
+  const testExe = test.cmd.at(0);
+  if (!testExe) {
+    return undefined;
+  }
+
+  // the meson target such that it is of meson type executable()
+  // and produces the binary that the test() executes.
+  const testDependencyTarget = targets.find((target) => {
+    const depend = test.depends.find((depend) => {
+      return depend == target.id && target.type == "executable";
+    });
+    return depend && testExe == target.filename.at(0);
+  });
+
+  // the first source file belonging to the target.
+  const path = testDependencyTarget?.target_sources
+    ?.find((elem) => {
+      return elem.sources;
+    })
+    ?.sources?.at(0);
+  return path ? vscode.Uri.file(path) : undefined;
+}
+
+/**
+ * ensures that the given test targets are up to date
+ * @param tests tests to rebuild
+ * @param buildDir meson buildDir
+ * @returns `ExecResult` of the `meson compile ...` invocation
+ */
```

Comment:
In all your comments (function descriptions, parameter descriptions), can you capital letters?